dataprofiler/profilers/float_column_profile.py

```diff
@@ -131,11 +142,9 @@ def _is_each_row_float(cls, df_series):
         :return: is_float_col
         :rtype: list
         """
-        len_df = len(df_series)
-        if len_df == 0:
-            return list()
-
-        return [NumericStatsMixin.is_float(x) for x in df_series]
+        if len(df_series) == 0: return list()
```

Comment:
Attempted: 
```python
query = r'^[\s]*[+-]?[\d]+[\.]*[\d]*[ ]?[*eE]*[ ]?[+-]?[\d]*[\s]*$'               
return df_series.str.count(query).astype(bool)
```

Ended up being slower overall. Method shown above.
```
         237106548 function calls (220371986 primitive calls) in 97.067 seconds

   Ordered by: internal time
   List reduced from 1199 to 50 due to restriction <50>

   ncalls  tottime  percall  cumtime  percall filename:lineno(function)
  942/858    9.873    0.010    9.874    0.012 {built-in method numpy.array}
 16716426    8.984    0.000    8.984    0.000 {method 'sub' of 're.Pattern' objects}
       12    7.280    0.607   23.221    1.935 {pandas._libs.lib.map_infer_mask}
 16716402    6.876    0.000   35.813    0.000 replace.py:122(re_replacer)
50170898/33454489    6.351    0.000   17.651    0.000 {built-in method builtins.isinstance}
 16716402    5.835    0.000    5.835    0.000 {method 'findall' of 're.Pattern' objects}
```

**Current method** (best) (in master)
```
         203668310 function calls (186933878 primitive calls) in 87.762 seconds

   Ordered by: internal time
   List reduced from 1200 to 50 due to restriction <50>

   ncalls  tottime  percall  cumtime  percall filename:lineno(function)
  912/858   11.158    0.012   11.159    0.013 {built-in method numpy.array}
 16716426    8.448    0.000    8.448    0.000 {method 'sub' of 're.Pattern' objects}
 16716402    6.824    0.000   34.672    0.000 replace.py:122(re_replacer)
50170088/33453679    6.200    0.000   17.103    0.000 {built-in method builtins.isinstance}
 16716409    5.395    0.000    7.831    0.000 typing.py:713(__subclasscheck__)
 16716409    4.521    0.000   12.351    0.000 typing.py:710(__instancecheck__)
       12    4.093    0.341    4.098    0.341 {pandas._libs.lib.map_infer}
 16716443    3.978    0.000    3.978    0.000 {method 'search' of 're.Pattern' objects}
       44    3.921    0.089    3.921    0.089 {built-in method builtins.sorted}
        6    3.528    0.588    6.278    1.046 float_column_profile.py:149(<listcomp>)

```